cluster/node-pools/master-default/userdata.yaml

```diff
@@ -693,3 +725,25 @@ write_files:
         - level: Request
           omitStages:
             - "RequestReceived"
+
+  - owner: root:root
+    path: /etc/kubernetes/config/encryption-config.yaml
+    permissions: '0400'
+    content: |
+      apiVersion: apiserver.config.k8s.io/v1
+      kind: EncryptionConfiguration
+      resources:
+        - resources:
+          - secrets
+          providers:
+      {{- if ne .Cluster.ConfigItems.enable_encryption "true" }}
+          - identity: {}
+      {{- end }}
+          - kms:
+              name: aws-encryption-provider
+              endpoint: unix:///var/run/kmsplugin/socket.sock
+              cachesize: 1000
+              timeout: 3s
+      {{- if eq .Cluster.ConfigItems.enable_encryption "true" }}
+          - identity: {}
+      {{- end }}
```

Comment:
This allows turning encryption on and off at will.

Reply:
We can't really turn it off once we've enabled it, can we?

Reply:
We can. All providers are used for reading but only the first is used for writing. This way one can seamless migrate from non-encrypted to encrypted, encrypted to non-encrypted and from one key to another.

The only thing to do is to force a write of every secret via `kubectl get secrets --all-namespaces -o json | kubectl replace -f -`.